http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamAuthProvider.scala

```diff
@@ -127,6 +127,52 @@ class SamAuthProvider[F[_]: OpenTelemetryMetrics](
     } yield (callerActions, projectCallerActions)
   }
 
+  /**
+   * Get all Sam resource IDs of the given type on which the user is granted permission to `discover`. Returns ids with explicitly
+   * granted roles only; users with workspace `owner` role, for example, can see all runtimes in the workspace, but
+   * `listResourceIds[SamRuntimeResourceId]` is not guaranteed to return all runtimes in that workspace. Works for any
+   * Sam resource type except SharedApp and similar alternative app types; these require a specific Sam resource object
+   * to determine permissions, as these are stored on the individual records.
+   * @param hasOwnerRole whether to return only resources that the user is granted ownership on (has role = `ownerRoleName`)
+   * @param userInfo the current user
+   * @param resourceDefinition the resource archetype (App, Runtime, etc)
+   * @param resourceIdDecoder converts Json -> SamResourceId, needed by SamDao
+   * @param ev the request context
+   * @tparam R SamResourceId
+   * @return List[SamResourceId] of all resources of type R which are discoverable (or owned, if `hasOwnerRole`) by the user
+   */
+  override def listResourceIds[R <: SamResourceId](
+    hasOwnerRole: Boolean,
+    userInfo: UserInfo
+  )(implicit
+    resourceDefinition: SamResource[R],
+    appDefinition: SamResource[AppSamResourceId],
+    resourceIdDecoder: Decoder[R],
+    ev: Ask[F, TraceId]
+  ): F[Set[R]] = {
+    require(resourceDefinition != appDefinition)
+    val authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token))
+
+    // TODO get `ownerRoleName` from Sam config for the resource type
+    val ownerRole: SamRole = resourceDefinition.ownerRoleName
+    val resourceType: SamResourceType = resourceDefinition.resourceType
```

Comment:
Seems this `val resourceType` isn't used -- it's passed implicitly via `SamResource` to `HttpSamDAO` I think.